### PR TITLE
Implement exercise to fhir transformer

### PR DESCRIPTION
**Pull request by brandonpollett**

### CODE REVIEW

File: Microsoft.HealthVault.Fhir/Transformers/ExerciseToFhir.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    /// <summary>
+    /// An extension class that transforms HealthVault exercise data types into FHIR Observations
+    /// </summary>
+    internal static class ExerciseToFhir
+    {
+        internal static Observation ToFhirInternal(Exercise exercise, Observation observation)
+        {
+            observation.Category = new List<CodeableConcept>() { FhirCategories.VitalSigns };
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/ExerciseToFhir.cs]: >observation.Category = new List<CodeableConcept>() { FhirCategories.VitalSigns }; [](start = 12, length = 81)

same as with sleep, this is not a vital sign. We should leave this empty

File: Microsoft.HealthVault.Fhir/Transformers/ObservationToExercise.cs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                        break;
+                    case HealthVaultExtensions.ExerciseDetail:
+                        ExerciseDetail exerciseDetail;
+                        var key = GetExerciseDetail(extension, out exerciseDetail);
+
+                        if (!string.IsNullOrEmpty(key))
+                        {
+                            exerciseSegment.Details.Add(key, exerciseDetail);
+                        }
+                        break;
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/ObservationToExercise.cs]: line breaks after closing braces.

File: Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        {
+            "url": "http://healthvault.com/fhir-extensions/thing-flags",
+            "valueString": "None"
+        },
+        {
+            "url": "http://healthvault.com/fhir-extensions/thing-state",
+            "valueString": "Active"
+        },
+        {
+            "url": "http://healthvault.com/extensions/exercise-detail",
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json]: >extensions [](start = 43, length = 10)

we are calling it fhir-extensions for some and extensions for some other, this looks wrong

File: Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+                            }
+                        ]
+                    }
+                },
+                {
+                    "url": "exercise-detail-value",
+                    "valueQuantity": {
+                        "value": 30,
+                        "unit": "seconds",
+                        "system": "http://healthvault.com/vocabularies/duration-units/1",
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json]: >1", [](start = 86, length = 3)

this 1 is strange, intended?

File: Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+            "extension": [
+                {
+                    "url": "exercise-segment-activity",
+                    "valueCodeableConcept": {
+                        "coding": [
+                            {
+                                "system": "http://healthvault.com/vocabularies/",
+                                "version": "1",
+                                "code": "exercise-activities:swimming",
+                                "display": "Swimming"
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json]: wouldn't it make more sense to be [http://healthvault.com/vocabularies/exercise-activities](http://healthvault.com/vocabularies/exercise-activities) as url and then code "swimming"?

File: Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+                },
+                {
+                    "url": "exercise-segment-offset",
+                    "valueDecimal": 43.3
+                },
+                {
+                    "url": "http://healthvault.com/extensions/exercise-detail",
+                    "extension": [
+                        {
+                            "url": "exercise-detail-name",
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirExercise.json]: > "url": "exercise-detail-name", [](start = 27, length = 31)

This is not a url

**Comment by brandonpollett** [comment]
@gcorvera I plan on opening a new PR once this is in where we make the handling of vocabularies and extension URLs consistent.

**Pull request merged**